File: 프로그래머스/lv2/42586. 기능개발/기능개발.cpp

#include <string>
#include <vector>
#include <queue>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    vector<int> day;
    for ( int i = 0; i < speeds.size(); i ++)
    {
        int temp = 100 - progresses[i];
        if ( temp % speeds[i] != 0 )
        {
            temp = temp / speeds[i] + 1;
        }else temp = temp / speeds[i];
        day.push_back(temp);
    }
    queue<int> q;
    for (int d:day)
    {
        if (q.empty())
        {
        q.push(d);
        }
        else
        {
            if (q.front() < d)
            {
                answer.push_back(q.size());
                while( !q.empty())
                {
                    q.pop();
                }
                q.push(d);
            }
            else 
            {
                q.push(d);
            }
        }
    }
    if (!q.empty()) answer.push_back(q.size());
    return answer;
}

File: 프로그래머스/lv2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    left = []
    queue = []
    for i in range(len(progresses)):
        tmp = 100-progresses[i]
        if tmp % speeds[i] == 0:
            tmp/=speeds[i]
        else:
            tmp = tmp // speeds[i] +1
            
        left.append(tmp)
        
    for i in left:
        if not queue:
            queue.append(i)
        elif queue[0] >= i:
            queue.append(i)
        elif queue[0] < i:
            answer.append(len(queue))
            queue = []
            queue.append(i)

    if queue:
        answer.append(len(queue))
        
    return answer